eng/Subsets.props

```diff
@@ -116,12 +102,12 @@
     <SubsetName Include="CoreHost" Description="The .NET Core hosts." />
     <SubsetName Include="Installer.NoCoreHost" Description="Hosting libraries, bundles, and installers. Includes these projects' tests." />
     <SubsetName Include="Installer.Managed" Description="The managed .NET hosting projects. This includes HostModel." />
-    <SubsetName Include="Installer.DepProjs" Description="The dependency projects. These gather shared framework files and run crossgen on them to turn them into ready-to-run (R2R) assemblies for the current platform." />
     <SubsetName Include="Installer.PkgProjs" Description="The packaging projects. These produce NETCoreApp assets: NuGet packages, installers, zips, and Linux packages." />
     <SubsetName Include="Bundles" Description="The shared framework bundle installer projects. Produces .exe installers for Windows." />
     <SubsetName Include="Installers" Description="Generates additional installers. This produces the shared frameworks and their installers." />
     <SubsetName Include="Installer.Tests" Description="The test projects. Note that building this doesn't execute tests: you must also pass the '-test' argument." />
     <SubsetName Include="Installer.Publish" OnDemand="true" Description="Builds the installer publish infrastructure." />
+    <SubsetName Include="sfx" OnDemand="true" Description="Builds the shared frameworks, bundles, installers with the new shared framework SDK." />
```

Comment:
Yes, it's supposed to represent "shared fx/shared framework". The reason I like a shorter name is developers need to type it out every time they want to build it (see why the CoreCLR subset is named clr and the Libraries subset is named libs).

cc: @ViktorHofer for any preference in naming (since I know there's plans to make the "shared framework" portion of the libraries build one subset)

Reply:
I was thinking of "framework" or "sharedframework". cc @jkotas and @ericstj for opinions.

Reply:
My understanding is that PR is just replacing an implementation detail of one of the existing subsets. Is it correct? If yes, what is value added by the new subset name?

How does one decide whether to build `Installer.PkgProjs`, `Bundles` or `sfx` ? All these look nearly identical to me.

Reply:
I need to remove the `Bundles` subset. That doesn't do anything any more.

The Pkgproj subset builds the `.pkgproj` projects under `src/installer` (so after this PR the RID-based NuGet packages for the hosting components).

The sfx subset builds the Nuget packs for Microsoft.NETCore.App.*, the zip/tar archives, and the installers for these components and the other installer projects for the host and hostfxr (including the bundle installers).

The installer subset only builds the installers for the Microsoft.NETCore.App, hostfxr, and the shared host, not the NuGet packages or the archives.

Reply:
Would it make sense to call it `Installer` or `Installer.Packs` ? Or just `packs` or `pkgs` if we wanted to drop `Installer` from the name?

I like the hierarchical naming that we use for clr or libs. It makes it easier to see how the different subsets relate to each other. Not sure how to apply it here.

Reply:
How's something like the following?

| Old subset name | New subset name |
|--------------------|---------------------|
| corehost              | host.native            |
| installer.pkgproj  | host.pkg                |
| sfx                       | packs                     |
| installers             | packs.installers       |

This moves the pkgproj subset under the hosting layer, which since all remaining projects in it are hosting related makes sense. It also moves the installers subset under the packs subset since it does a subset of the work that the packs subset does. Then we'd have a `host` subset that builds `host.native` and `host.pkg`.

And then the overall layering of the product would be 

- clr has no dependencies
- mono has no dependencies
- host has no dependencies
- libs depends on either clr or mono (once #38034 then libs would also depend on host)
- packs depends on libs, host, and either clr or mono.

Reply:
- Curious, who depends on the host packages besides ourselves?
- Do we need separate subsets for the host native build and its packages?
- Do we need separate subsets for "packs" and "packs.installlers"? Who consumes `packs.installers`?

Reply:
> * Curious, who depends on the host packages besides ourselves?

I'm not sure, but since we publicly ship them, I didn't want to unilaterally remove them.

> * Do we need separate subsets for the host native build and its packages?

I think it's worth having a separate subset. They only need to be built so we can ship them. When the AppModel team builds the hosts, they don't need to build the packages to run tests locally so I'd like to avoid adding extra steps in their workflow during this cleanup (which is significantly helping their workflow)

> * Do we need separate subsets for "packs" and "packs.installlers"? Who consumes `packs.installers`?

Currently, the installer CI build uses special docker images with the deb and rpm tools installed to build the deb and rpm packages. CI uses `packs.installers` subset to directly call the `GenerateInstallers` target on each installer project easily.

Reply:
@jkoritzinsky I like your proposal.

> Curious, who depends on the host packages besides ourselves?

I assume that you are talking about `microsoft.netcore.app.host.*` packages.
SDK uses them. For example, run this on Windows:
```
dotnet new console
dotnet publish --no-self-contained -r linux-x64 /v:diag
```
And look for `.nuget\packages\microsoft.netcore.app.host.linux-x64\5.0.0-rc.1.20451.14\runtimes\linux-x64\native\apphost` in the output.

Reply:
The host packages we’re referring to are the following packages:

- Microsoft.NETCore.DotNetHost
- Microsoft.NETCore.DotNetHostPolicy
- Microsoft.NETCore.DotNetHostResolver

The package you mentioned is produced as a special framework pack and would be part of the packs subset.

Reply:
I had some more time to work on this PR, so I've implemented the new subset names from this discussion. I'll update the docs and the PR description next.